### PR TITLE
Guaranteed income calculator

### PR DESCRIPTION
**Pull request by benbarnett**
Questions:

* [x] Do we restrict by age range? We currently limit 55-75.

### CODE REVIEW

File: app/forms/calculators/guaranteed_income_form.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+require 'active_model'
+
+module Calculators
+  class GuaranteedIncomeForm
+    include ActiveModel::Model
+
+    attr_reader :pot, :age
+
+    validates :pot, presence: true, numericality: { allow_blank: true, greater_than: 0 }
+    validates :age, presence: true, numericality: { allow_blank: true, greater_than: 54, less_than: 76 }
```
Comment (andrewgarner) [inline app/forms/calculators/guaranteed_income_form.rb]: `54` and `76` are magic numbers. We should extract them to constants that reveal the significance of those numbers.

File: app/views/calculators/guaranteed_income/_estimate.html.erb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  and get:
+
+  <strong class="calculator__estimate__number t-calculator-income">
+    <%= number_to_currency(estimate[:income]) %> guaranteed annual taxable income, for the rest of your life
+  </strong>
+</p>
+
+<ul class="t-calculator-notes">
+  <li>
+    The amount in your pot will be affected by inflation and any fees your provider charges.
```
Comment (andrewgarner) [inline app/views/calculators/guaranteed_income/_estimate.html.erb]: There isn't anything left in the pot?

File: app/views/calculators/guaranteed_income/_estimate.html.erb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<p>
+  Based on what you’ve told us, if you take your <%= number_to_currency(pot) %> pension pot as an annuity, you could get:
```
Comment (andrewgarner) [inline app/views/calculators/guaranteed_income/_estimate.html.erb]: Do you take it as an annuity or do you use it to buy an annuity? @guidance-guarantee-programme/content-designers

File: app/views/calculators/guaranteed_income/_form.html.erb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+<h2 id="calculator">Estimate how much your guaranteed income could be</h2>
+
+<%= render 'calculators/error_summary', messages: errors&.messages if form&.invalid? %>
+
+<form action="/guaranteed-income/estimate#form" method="get">
+  <%= render 'calculators/form_input_currency',
+             field: :pot,
+             value: form&.pot,
+             label: 'Amount you\'ll have your pot when you retire',
```
Comment (andrewgarner) [inline app/views/calculators/guaranteed_income/_form.html.erb]: `'Amount you’ll have your pot when you retire'`

Comment (benbarnett) [inline app/views/calculators/guaranteed_income/_form.html.erb]: This is all placeholder copy.

Comment (andrewgarner) [inline app/views/calculators/guaranteed_income/_form.html.erb]: Do you have to retire to buy an annuity?

Maybe just "Amount you’ll have in your pot" or shall we assume that the customer is in a position to buy an annuity now in which case we could ask "Amount in your pot" as we do for the other calculators.

@guidance-guarantee-programme/content-designers

File: config/routes.rb [hunk 5]
```diff
@@ -16,6 +16,7 @@
     get 'leave-pot-untouched/estimate', to: 'calculators/leave_pot_untouched#show'
     get 'take-cash-in-chunks/estimate', to: 'calculators/take_cash_in_chunks#show'
     get 'take-whole-pot/estimate', to: 'calculators/take_whole_pot#show'
+    get 'guaranteed-income/estimate', to: 'calculators/guaranteed_income#show'
```
Comment (andrewgarner) [inline config/routes.rb]: If it were me I'd keep things ordered alphabetically - because I'm that sad!

File: features/calculators.feature [hunk 6]
```diff
@@ -12,6 +12,7 @@ Feature: Calculators
       | leave-pot-untouched |
       | take-cash-in-chunks |
       | take-whole-pot      |
+      | guaranteed-income   |
```
Comment (andrewgarner) [inline features/calculators.feature]: Again on ordering!

Comment (benbarnett) [inline features/calculators.feature]: Will fix!

Comment (andrewgarner) [inline features/calculators.feature]: :+1:

File: app/forms/calculators/guaranteed_income_form.rb (hunk 1 above)
Comment (benbarnett) [inline app/forms/calculators/guaranteed_income_form.rb]: Extracted out

File: app/views/calculators/guaranteed_income/_estimate.html.erb (hunk 3 above)
Comment (alexkouz) [inline app/views/calculators/guaranteed_income/_estimate.html.erb]: The 2nd one, so:

'Based on what you’ve told us, if you use your <%= number_to_currency(pot) %> pension pot to buy an annuity, you could get:'

File: app/views/calculators/guaranteed_income/_form.html.erb (hunk 4 above)
Comment (alexkouz) [inline app/views/calculators/guaranteed_income/_form.html.erb]: "...or shall we assume that the customer is in a position to buy an annuity now in which case we could ask "Amount in your pot" as we do for the other calculators."

This!

Comment (benbarnett) [inline app/views/calculators/guaranteed_income/_form.html.erb]: Great thanks, will update

File: app/views/calculators/guaranteed_income/_estimate.html.erb (hunk 3 above)
Comment (benbarnett) [inline app/views/calculators/guaranteed_income/_estimate.html.erb]: Thanks, have updated

**Comment by benlovell** [comment]
Most of the comments I raised in #472 would also apply here. There also appears to be a reasonable amount of incidental duplication occurring in the structure of these calculators (mostly the controllers and views). Perhaps this should be addressed after both this PR and #472 are merged though, and shouldn't be a blocker for now.

**Comment by benbarnett** [comment]
@benlovell I think I've addressed most of the stylistic comments now

**Comment by benlovell** [comment]
@benbarnett One thing to consider is the fragility of cucumber steps that assert on whole blocks of content, as we discussed with the other calculator.

**Pull request merged**